msv/src/main/java/com/sun/msv/reader/GrammarReader.java

```diff
@@ -245,13 +246,13 @@ public String[] splitQName( String qName ) {
             // if the default namespace is not bounded, return "".
             // this behavior is consistent with SAX.
             if(ns==null)    ns="";
-            return new String[]{ns,qName,qName};
+            return new String[]{ns, qName, ns};
```

Comment:
but the prefix here is "null", not "ns"?

Reply:
The prefix is not existent and I would go for empty string (to play it safe) but it is not ns, indeed! Thanks, Michael!
(I have added a new commit with this fix and some further consistent prefix handling)